doc/kernel/memory_management/shared_multi_heap.rst

```diff
@@ -5,80 +5,78 @@ Shared Multi Heap
 
 The shared multi-heap memory pool manager uses the multi-heap allocator to
 manage a set of reserved memory regions with different capabilities /
-attributes (cacheable, non-cacheable, etc...) defined in the DT.
-
-The user can request allocation from the shared pool specifying the capability
-/ attribute of interest for the memory (cacheable / non-cacheable memory,
-etc...).
-
-The different heaps with their attributes available in the shared pool are
-defined into the DT file leveraging the ``reserved-memory`` nodes.
-
-This is a DT example declaring three different memory regions with different
-cacheability attributes: ``cacheable`` and ``non-cacheable``
-
-.. code-block:: devicetree
-
-   / {
-        reserved-memory {
-                compatible = "reserved-memory";
-                #address-cells = <1>;
-                #size-cells = <1>;
-
-                res0: reserved@42000000 {
-                        compatible = "shared-multi-heap";
-                        reg = <0x42000000 0x1000>;
-                        capability = "cacheable";
-                        label = "res0";
-                };
-
-                res1: reserved@43000000 {
-                        compatible = "shared-multi-heap";
-                        reg = <0x43000000 0x2000>;
-                        capability = "non-cacheable";
-                        label = "res1";
-                };
-
-                res2: reserved2@44000000 {
-                        compatible = "shared-multi-heap";
-                        reg = <0x44000000 0x3000>;
-                        capability = "cacheable";
-                        label = "res2";
-                };
-        };
-
-The user can then request 4K from heap memory ``cacheable`` or
-``non-cacheable`` using the provided APIs:
+attributes (cacheable, non-cacheable, etc...).
+
+All the different regions can be added at run-time to the shared multi-heap
+pool providing an opaque "attribute" value (an integer or enum value) that can
+be used by drivers or applications to request memory with certain capabilities.
+
+This framework is commonly used as follow:
+
+1. At boot time some platform code initialize the shared multi-heap framework
+   using :c:func:`shared_multi_heap_pool_init()` and add the memory regions to
+   the pool with :c:func:`shared_multi_heap_add()`, possibly gathering the
+   needed information for the regions from the DT.
+
+2. Each memory region encoded in a :c:type:`shared_multi_heap_region`
+   structure.  This structure is also carrying an opaque and user-defined
+   integer value that is used to define the region capabilities (for example:
+   cacheability, cpu affinity, etc...)
+
+.. code-block:: c
+
+   // Init the shared multi-heap pool
```

Comment:
I think we need to avoid C99 comments.

Reply:
Yeah, I mean, this is a documentation file, so maybe not that important. Anyway, I wait for more comments before fixing this.